bio_hansel/scripts/find_cluster.py

```diff
@@ -0,0 +1,100 @@
+from typing import List, Dict
+
+import numpy as np
+import pandas as pd
+import scipy as sp
+
+from scipy.cluster.hierarchy import fcluster, linkage
+from scipy.spatial.distance import pdist
+
+
+def find_clusters(df: pd.DataFrame, min_threshold: float,
+                  max_threshold: float) -> Dict[str, str]:
+    """
+    Takes in a vcf file and creates clusters from the scipy hierarchy clustering algorithm
+    
+    Example:
+    '/path/example.vcf' -> {'mysnpsSRR2323':'1', 'mysnpsSRR232323':'2'}
+    
+    Args:
+        df: contains the vcf information in the DataFrame format
+        min_threshold: minimum threshold for the flat cluster array
+        max_threshold: maximum threshold for the flat cluster array
+
+    Returns:
+        cluster_dict: a dictionary indicating the cluster membership of each of the supplied genomes in
+                the vcf file
+    """
+
+    distance_matrix = compute_distance_matrix(df)
+    clustering_array = create_linkage_array(distance_matrix)
+
+    flat_clusters = output_flat_clusters(clustering_array, df.columns,
+                                         min_threshold, max_threshold)
+
+    subset = flat_clusters.iloc[0]
+    cluster_dict = subset.to_dict()
+    return cluster_dict
+
+
+def compute_distance_matrix(df: pd.DataFrame) -> List:
+    """Takes in a binary SNV state DataFrame and outputs a distance matrix using the hamming method
+
+    Args:
+        df: the DataFrame that contains only the samples' binary data
+
+    Returns:
+        distance_matrix: an matrix of pair-wise distances between samples
+    """
+    distance_matrix = sp.spatial.distance.pdist(
```

Comment:
In cases where you are returning the value right away on the next line, it's best not to introduce a local variable.

Reply:
yup, this has been addressed in commit 5c6f81152eb37d0bc371b08673de4241824d3803